proxy-robots-module/src/main/java/orwell/proxy/UdpBeaconFinder.java

```diff
@@ -0,0 +1,145 @@
+package orwell.proxy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.*;
+import java.util.Enumeration;
+
+/**
+ * Created by Michaël Ludmann on 5/25/15.
+ */
+public class UdpBeaconFinder {
+    private final static Logger logback = LoggerFactory.getLogger(UdpBeaconFinder.class);
+    private final int broadcastPort;
+    private final int receiverBufferSize = 512;
+    private final DatagramSocket datagramSocket;
+    private final UdpBeaconDecoder udpBeaconDecoder;
+    private int maxAttemptsNumber = 1;
+    private int attemptsPerformed = 0;
+    private String pushAddress;
+    private String subscribeAddress;
+
+    public UdpBeaconFinder(final DatagramSocket datagramSocket, final int broadcastPort, final UdpBeaconDecoder udpBeaconDecoder) {
+        this.datagramSocket = datagramSocket;
+        this.broadcastPort = broadcastPort;
+        this.udpBeaconDecoder = udpBeaconDecoder;
+    }
+
+    public void setMaxAttemptsNumber(final int maxAttemptsNumber) {
+        if (1 > maxAttemptsNumber) {
+            logback.warn("Udp broadcasting attempts number cannot be less than 1");
+        } else {
+            this.maxAttemptsNumber = maxAttemptsNumber;
+        }
+    }
+
+    public int getNumberOfPerformedAttempts() {
+        return attemptsPerformed;
+    }
+
+    /**
+     * @return true if we did not get any meaningful response from server
+     * and have not reached max allowed attempts number
+     */
+    private boolean shouldTryToFindBeacon() {
+        if (null == udpBeaconDecoder) {
+            return attemptsPerformed < maxAttemptsNumber;
+        }
+        return (!udpBeaconDecoder.hasReceivedCorrectData() &&
+                attemptsPerformed < maxAttemptsNumber);
+    }
+
+    /**
+     * Find the server using UDP broadcast and fill data fields
+     */
+    public void startBroadcasting() {
```

Comment:
This does more than start to broadcast. You may want to change the name and split into more methods.